src/bundle/Resources/public/js/scripts/core/dropdown.js

```diff
@@ -39,8 +39,8 @@
             this.itemsListContainer = this.itemsContainer.querySelector('.ibexa-dropdown__items-list');
             this.itemsFilterInput = this.itemsContainer.querySelector('.ibexa-dropdown__items-filter');
 
-            this.hasDefaultSelection = config.hasDefaultSelection ?? false;
             this.canSelectOnlyOne = !this.sourceInput?.multiple;
+            this.hasDefaultSelection = config.hasDefaultSelection ?? this.canSelectOnlyOne;
```

Comment:
Here I would put `||` instead of `??`, imo `??` should be used only when second argument is constant

Reply:
I wouldn't subject the use of `??` to the type of argument used after it. Whether the last argument is constant or not it does not affect `??` behaviour.

When it comes to discussing behaviour, in your case, given that `config.hasDefaultSelection` was set to `false` it would default to `this.canSelectOnlyOne`. Do we want such behaviour?

Reply:
I think the logic here is correct.
Not set -> default
set false -> false